codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java

```diff
@@ -133,18 +133,15 @@ public String blobShape(final BlobShape shape) {
       : dataSource;
     return """
     func () []byte {
-    b := []byte{}
-    if %s == nil {
-        return nil
-    }
-    for i := dafny.Iterate(%s) ; ; {
-        val, ok := i()
-        if !ok {
-            return b
-        } else {
-            b = append(b, val.(byte))
-        }
-    }
+      if %s == nil {
+          return nil
+      }
+      a := %s.(dafny.Sequence).ToArray().(dafny.GoNativeArray);
```

Comment:
The dataSource could also contain assertion. I know its very confusing so it was very hard to create a codegen for Go.

```suggestion
      a := %s.ToArray().(dafny.GoNativeArray);
```

Reply:
> The dataSource could also contain assertion.

Thats why in 131 it says `dataSource.startsWith("input.(")?...`